lib/normalize.js

```diff
@@ -862,272 +1092,168 @@ const generateBorderPositionShorthand = (items, property, prior) => {
-        } else {
-          lineStyleItems.set(property, { property, value, priority });
-        }
-        break;
-      }
-      case "color": {
-        if (priority) {
-          lineColorPriorItems.set(property, { property, value, priority });
-        } else {
-          lineColorItems.set(property, { property, value, priority });
-        }
-        break;
-      }
-      default:
-    }
-    switch (positionPart) {
-      case "top": {
-        if (priority) {
-          positionTopPriorItems.set(property, { property, value, priority });
-        } else {
-          positionTopItems.set(property, { property, value, priority });
-        }
-        break;
-      }
-      case "right": {
-        if (priority) {
-          positionRightPriorItems.set(property, { property, value, priority });
-        } else {
-          positionRightItems.set(property, { property, value, priority });
-        }
-        break;
-      }
-      case "bottom": {
-        if (priority) {
-          positionBottomPriorItems.set(property, { property, value, priority });
-        } else {
-          positionBottomItems.set(property, { property, value, priority });
-        }
-        break;
-      }
-      case "left": {
-        if (priority) {
-          positionLeftPriorItems.set(property, { property, value, priority });
-        } else {
-          positionLeftItems.set(property, { property, value, priority });
-        }
-        break;
-      }
-      default:
-    }
-  }
-  if (lineWidthItems.size === 4) {
-    const [property, item] = generateBorderLineShorthand(lineWidthItems, "border-width") ?? [];
-    if (property && item) {
-      properties.set(property, item);
-    }
-  } else if (lineWidthPriorItems.size === 4) {
-    const [property, item] =
-      generateBorderLineShorthand(lineWidthPriorItems, "border-width", "important") ?? [];
-    if (property && item) {
-      properties.set(property, item);
-    }
-  }
-  if (lineStyleItems.size === 4) {
-    const [property, item] = generateBorderLineShorthand(lineStyleItems, "border-style") ?? [];
-    if (property && item) {
-      properties.set(property, item);
-    }
-  } else if (lineStylePriorItems.size === 4) {
-    const [property, item] =
-      generateBorderLineShorthand(lineStylePriorItems, "border-style", "important") ?? [];
-    if (property && item) {
-      properties.set(property, item);
-    }
-  }
-  if (lineColorItems.size === 4) {
-    const [property, item] = generateBorderLineShorthand(lineColorItems, "border-color") ?? [];
-    if (property && item) {
-      properties.set(property, item);
-    }
-  } else if (lineColorPriorItems.size === 4) {
-    const [property, item] =
-      generateBorderLineShorthand(lineColorPriorItems, "border-color", "important") ?? [];
-    if (property && item) {
-      properties.set(property, item);
-    }
+const borderCollections = {
+  [WIDTH]: {
+    shorthand: BORDER_WIDTH,
+    generator: generateBorderLineShorthand,
+    items: new Map(),
```

Comment:
This module-level mutable state is worrying. Will it cause problems if multiple jsdom instances use it? At the very least it needs good documentation.

Reply:
https://github.com/jsdom/cssstyle/pull/261/commits/66437d1d888a3124529363599d5021d30a693bd7

Please take a look.